backend/src/Service/Subscription/SubscriptionService.php

```diff
@@ -33,105 +31,246 @@ public function __construct(AutoMapping $autoMapping, SubscriptionManager $subsc
      */
     public function createSubscription(SubscriptionCreateRequest $request): mixed
     {
-        $isFuture = $this->getIsFuture($request->getStoreOwner());
-        if ( $isFuture === 0) {
+        $this->checkSubscription($request->getStoreOwner());
+       
+        $isFuture = $this->getIsFutureState($request->getStoreOwner());
 
-            $status = SubscriptionConstant::SUBSCRIBE_INACTIVE;
+        $request->setIsFuture($isFuture);
 
-            $subscriptionCurrent = $this->getSubscriptionCurrent($request->getStoreOwner());
-  
-            if($subscriptionCurrent) {
+        $subscription = $this->subscriptionManager->createSubscription($request);
 
-                $status = $this->subscriptionIsActive($subscriptionCurrent['id']);
-            }
+        return $this->autoMapping->map(SubscriptionEntity::class, SubscriptionResponse::class, $subscription);
+    }
 
-            $subscriptionResult = $this->subscriptionManager->createSubscription($request, $status);
+    /**
+     * Check if there is a future subscription
+     * @param storeOwner
+     * @return INT
+     */
+    public function getIsFutureState($storeOwner): INT
+    {
+        $subscriptionCurrent = $this->subscriptionManager->getSubscriptionCurrent($storeOwner);
+        if($subscriptionCurrent) {
+            if($subscriptionCurrent->getStatus() === SubscriptionConstant::SUBSCRIBE_ACTIVE) {
+             
+                return  1;
+            }
 
-            return $this->autoMapping->map(SubscriptionEntity::class, SubscriptionResponse::class, $subscriptionResult);
+            return  0;
```

Comment:
we need to use const instead of string or numbers.

Reply:
done